File: PHP/8php 3/positive_sum.php
<?php
/*
|--------------------------------------------------------------------------
| Sum of positive
|--------------------------------------------------------------------------
|
*/

function positive_sum($arr) {
   $filter = array_filter($arr, function($i) { return $i >= 0; });
   return count($filter) == 0 ? 0 : array_sum($filter);
}


function positive_sum($arr) {
   $sum = 0;
   foreach ($arr as $element) {
      $sum += $element > 0 ? $element : 0;
   }
   return $sum;
}


function positive_sum($a) {
   return array_sum(array_filter($a, function ($n) {return $n > 0;}));
}


function positive_sum($arr) {
   if (preg_replace('/(-\d.*)/','0',$arr)){
   $rpl = preg_replace('/(-\d.*)/','0',$arr);
   return array_sum($rpl);}else{
   return array_sum($arr);}
}


echo(positive_sum([1, 2, 3, 4, 5])), '
';
echo(positive_sum([1, -2, 3, 4, 5])), '
';
echo(positive_sum([])), '
';

File: Python/8py 2/positive_sum.py
"""
|--------------------------------------------------------------------------
| Sum of positive
|--------------------------------------------------------------------------
|
"""

def positive_sum(arr):
   res = 0
   for i in range(len(arr)):
      if arr[i] > 0:
         res += arr[i]
   return res


positive_sum = lambda a: sum(e for e in a if e > 0)


def positive_sum(arr):
   return sum(x for x in arr if x > 0)


def positive_sum(arr):
   return sum(filter(lambda x: x > 0,arr))


def positive_sum(arr):
   return sum(map(lambda x: x if x > 0 else 0, arr))


def positive_sum(arr):
   return sum( max(i, 0) for i in arr )


print(positive_sum([1,2,3,4,5]),15)
print(positive_sum([1,-2,3,4,5]),13)
print(positive_sum([-1,2,3,4,-5]),9)